libtransact/src/contract/address/double_key_hash.rs

```diff
@@ -48,6 +52,35 @@ impl Addresser<(String, String)> for DoubleKeyHashAddresser {
     }
 }
 
+// Used to validate the lengths of the input, including the prefix length and optional custom hash
+// length, which are used to construct the DoubleKeyHashAddresser.
+fn validate_lengths(
+    prefix_length: usize,
+    first_length: Option<usize>,
+) -> Result<(), AddresserError> {
+    // Validate the length of the prefix is not greater than the ADDRESS_LENGTH.
+    if prefix_length > ADDRESS_LENGTH {
+        return Err(AddresserError {
+            message: format!("Prefix length is greater than {}", ADDRESS_LENGTH),
+        });
+    }
+
+    // Validate the length of the prefix plus the custom hash length is not greater than the
+    // ADDRESS_LENGTH.
+    if let Some(custom_length) = first_length {
+        if custom_length + prefix_length > ADDRESS_LENGTH {
+            return Err(AddresserError {
+                message: format!(
+                    "Custom hash length and length of prefix combined are greater than {}",
```

Comment:
We should avoid the "custom hash length" terminology.

Suggest: "Length of prefix ({}) plus length of first address segment ({}) combined are greater than total address length ({})"